src/cpp/client/client_interceptor.cc

```diff
@@ -28,7 +28,17 @@ experimental::ClientInterceptorFactoryInterface*
 namespace experimental {
 void RegisterGlobalClientInterceptorFactory(
     ClientInterceptorFactoryInterface* factory) {
+  if (internal::g_global_client_interceptor_factory != nullptr) {
+    GPR_ASSERT(false &&
+               "It is illegal to call RegisterGlobalClientInterceptorFactory "
+               "multiple times.");
+  }
   internal::g_global_client_interceptor_factory = factory;
 }
+
+// For testing purposes only.
+void TestOnlyResetGlobalClientInterceptorFactory() {
```

Comment:
I think it's ok to declare this in the .h file. The TestOnly bit should make it clear that it's not for use. You could even make it namespace grpc::testing like we use in ServerContextTestSpouse.

Reply:
done